### DISCUSSION TITLE
Everything could be more interesting

### DISCUSSION THREAD

**Discussion by unclegluk** [hunar4321/particle-life#26] [General]
If only it were done in Unity.

**Comment by redblobgames** [comment]
Watching the video, it's *very little* code, and I would encourage you to write it in Unity or any other system! :-)

**Comment by hunar4321** [comment]
I'm not good with Unity 3d. I tried it before, it is very slow so maybe those who know Unity better can step in and make a 3d version of it.

**Comment by AnasQiblawi** [comment]
Hi, I was thinking, how about making folders for each Programing language,
eg. 
- C++
- JavaScript 
- Python

so it's easier for new people and also it will give more space for each language to write more files in future if needed.

**Reply by hunar4321** [reply]
Thanks for the suggestion, I tried to move them to separate folders but it removed the contribute list, therefore, I reverted back. Also, I realized if do that, many links out there will become broken so for now I'll leave it like that. I think it won't be confusing as long as all the code are kept in one file per langauge

**Reply by redblobgames** [reply]
Hunar, my (admittedly unsolicited) advice:

Decide if *you* want to be managing unity, python, godot, etc. open source projects, especially reviewing code in languages you don't use. If you don't, then it'd be simpler for the `hunar4321/particle-life` project to be *your* project with c++ and js, and then *link* to unity, godot, python, and other versions. Then you don't have to spend the time reviewing the code submissions for languages you don't use.

For an example of this, https://github.com/mapbox/delaunator is a javascript project, and the author links to projects in other languages (rust, c++, c#, python, etc.). This is also better for people *using* that project, because the people wanting to contribute python don't have to worry about the c++ code mixed in the same project, etc.

**Reply by hunar4321** [reply]
@redblobgames 
That is actually a great advice. 
I'm definitely not a hardcore programmer and also not planning to take this project far beyond it's educational purpose. 
So yes, that is a great idea Thanks!

**Comment by hunar4321** [comment]
> If only it were done in Unity.

By the way, someone has already ported the code to Godot Game Engine. Check his work out here. I'm sure people will soon do it in Unity as well. 

https://youtu.be/f4BXY_vp4f8

**Comment by ker2x** [comment]
I worked really hard on optimization of the C++ version, a unity version would be significantly slower.
I could make a faster version, using intel oneAPI, but it would reduce portability.
